test/distributed/org/apache/cassandra/distributed/test/SSTableIdGenerationTest.java

```diff
@@ -0,0 +1,515 @@
+     * both the current and legacy sstable activity tables are updated.
+     * Then, when enable uuid, we actually create sstables with uuid but keep and can read the old sstables. Also, only
+     * update the current sstable activity table.
+     */
+    @Test
+    public void testRestartWithUUIDEnabled() throws IOException
+    {
+        try (Cluster cluster = init(Cluster.build(1)
+                                           .withDataDirCount(1)
+                                           .withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, false))
+                                           .start()))
+        {
+            cluster.schemaChange(createTableStmt(KEYSPACE, "tbl", null));
+            createSSTables(cluster.get(1), KEYSPACE, "tbl", 1, 2);
+            assertSSTablesCount(cluster.get(1), 2, 0, KEYSPACE, "tbl");
+            verfiySSTableActivity(cluster, true);
+
+            restartNode(cluster, 1, true);
+
+            createSSTables(cluster.get(1), KEYSPACE, "tbl", 3, 4);
+            assertSSTablesCount(cluster.get(1), 2, 3, KEYSPACE, "tbl");
+            verfiySSTableActivity(cluster, false);
+
+            checkRowsNumber(cluster.get(1), KEYSPACE, "tbl", 9);
+        }
+    }
+
+    /**
+     * This test verifies that we should not be able to start a node with uuid disabled when there are uuid sstables
+     */
+    @Test
+    public void testRestartWithUUIDDisabled() throws IOException
+    {
+        try (Cluster cluster = init(Cluster.build(1)
+                                           .withDataDirCount(1)
+                                           .withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, true))
+                                           .start()))
+        {
+            cluster.disableAutoCompaction(KEYSPACE);
+            cluster.schemaChange(createTableStmt(KEYSPACE, "tbl", null));
+            createSSTables(cluster.get(1), KEYSPACE, "tbl", 1, 2);
+            assertSSTablesCount(cluster.get(1), 0, 2, KEYSPACE, "tbl");
+            verfiySSTableActivity(cluster, false);
+
+            Assertions.assertThatExceptionOfType(RuntimeException.class)
+                      .isThrownBy(() -> restartNode(cluster, 1, false))
+                      .withCauseInstanceOf(SystemExitException.class);
+        }
+    }
+
+    @Test
+    public final void testCompactionStrategiesWithMixedSSTables() throws Exception
+    {
+        testCompactionStrategiesWithMixedSSTables(SizeTieredCompactionStrategy.class,
+                                                  DateTieredCompactionStrategy.class,
+                                                  TimeWindowCompactionStrategy.class,
+                                                  LeveledCompactionStrategy.class);
+    }
+
+    /**
+     * The purpose of this test is to verify that we can compact using the given strategy the mix of sstables created
+     * with sequential id and with uuid. Then we verify whether the number of rows.
```

Comment:
Nit: `Then we verify whether the number of rows.` seems like an unfinished sentence, maybe it's just `Then we verify the number of rows.`?